### ISSUE TITLE
Emit with name-colliding helper functions is incorrect when target >=ES2015

### ISSUE THREAD

**Issue by joeywatts** [microsoft/TypeScript#33547]
**TypeScript Version:**  3.6.3

**Search Terms:** helper rename

**Code**

```ts
// @target: es2015

declare function __awaiter(): void;
class C {
    async func() {
    }
}
```

**Expected behavior:**
Some sort of error is emitted (when the target requires the helper) notifying the user that the generated code will not work OR maybe the helper is emitted with an alternative name.

**Actual behavior:**
The emit declares an `__awaiter` function, but attempts to call `__awaiter_1`.

**Playground Link:** [here](http://www.typescriptlang.org/play/index.html?target=2#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXwH0CoB3KLDEGACgEoAueANxy2AG4AoSKAZ1-gBheAG9O8CfD4BPdEjRg6o8ZIC+nVUA)

**Related Issues:**

**Comment by RyanCavanaugh** [comment]
Fix: Add more underscores 😅

**Comment by srclark213** [comment]
Is there any news on this? I ran into a similar issue when an npm package I pulled in already had the helper `__awaiter` defined so calls changed to `__awaiter_1` but no `__awaiter_1` function ever got defined. Had to write a hack post-build script to set `__awaiter_1 = __awaiter` after compilation, would be great if this got fixed.

**Comment by rbuckton** [comment]
This looks to be a duplicate of #33269, which was fixed by #36541.
